ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListQueriesExecutor.java

```diff
@@ -168,43 +181,55 @@ private static void mergeSimple(
                 query,
                 Collections.singletonMap(
                     new KsqlHostInfoEntity(sessionProperties.getKsqlHostInfo()),
-                    query.getState()))));
+                    KsqlConstants.fromStreamsState(
+                        KafkaStreams.State.valueOf(query.getState()))
+                ))));
   }
 
   private static void mergeExtended(
-      final List<KsqlEntity> remoteResults,
+      final Pair<List<KsqlEntity>, Set<HostInfo>> remoteResults,
       final Map<QueryId, QueryDescription> allResults
   ) {
-    final List<QueryDescription> remoteQueryDescriptions = remoteResults.stream()
+    final List<KsqlEntity> remoteQueries = remoteResults.getLeft();
+    final List<QueryDescription> remoteQueryDescriptions = remoteQueries.stream()
         .map(QueryDescriptionList.class::cast)
         .map(QueryDescriptionList::getQueryDescriptions)
         .flatMap(List::stream)
         .collect(Collectors.toList());
     for (QueryDescription q : remoteQueryDescriptions) {
       final QueryId queryId = q.getId();
 
-      // If the query has already been discovered, add to the ksqlQueryHostState mapping
+      // If the query has already been discovered, add to the ksqlQueryHostStatus mapping
       if (allResults.containsKey(queryId)) {
-        for (Map.Entry<KsqlHostInfoEntity, String> entry :
-            q.getKsqlHostQueryState().entrySet()) {
+        for (Map.Entry<KsqlHostInfoEntity, KsqlQueryStatus> entry :
+            q.getKsqlHostQueryStatus().entrySet()) {
           allResults
               .get(queryId)
-              .updateKsqlHostQueryState(entry.getKey(), entry.getValue());
+              .updateKsqlHostQueryStatus(entry.getKey(), entry.getValue());
         }
       } else {
         allResults.put(queryId, q);
       }
     }
+    
+    final Set<HostInfo> unresponsiveRemoteHosts = remoteResults.getRight();
+    for (HostInfo hostInfo: unresponsiveRemoteHosts) {
+      for (QueryDescription queryDescription: allResults.values()) {
+        queryDescription.updateKsqlHostQueryStatus(
+            new KsqlHostInfoEntity(hostInfo.host(), hostInfo.port()),
+            KsqlQueryStatus.UNRESPONSIVE);
+      }
+    }
   }
 
-  private static List<KsqlEntity> scatterGather(
+  private static Pair<List<KsqlEntity>, Set<HostInfo>> scatterGather(
```

Comment:
It would be better if CompletableFuture was used here instead of the legacy blocking Future class.
That way we wouldn't have to refactor this code when we move to a non blocking model.
This is a general point really, we shouldn't be using Future in new code.

Reply:
I can open another PR to change it to use CompletableFuture